daemon/containerd/image_exporter.go

```diff
@@ -248,8 +248,21 @@ func (i *ImageService) LoadImage(ctx context.Context, inTar io.ReadCloser, platf
 			if nameFromArchive == "" {
 				return false
 			}
-			_, err := reference.ParseNormalizedNamed(nameFromArchive)
-			return err == nil
+
+			ref, err := reference.ParseNormalizedNamed(nameFromArchive)
+			if err != nil {
+				return false
+			}
+
+			// Look up if there is an existing image with this name and ensure a dangling image exists.
+			if img, err := i.resolveImage(ctx, ref.String()); err != nil && !errdefs.IsNotFound(err) {
```

Comment:
We can simplify `resolveImage` to just `i.images.Get(ctx, ref.String())` because we already have the full reference and don't need to do any fuzzy matching.